### PR TITLE
fix: run node-gyp rebuild when install is not specified

### PR DESCRIPTION
**Pull request by etamponi**
The intent was there before, but the code assumed node-gyp was in the `node_modules` under `supi`, which might not be the case because of dependency deduping. Instead of doing path incantations, I just rewrite `script['install']` to equal the default https://docs.npmjs.com/misc/scripts

### CODE REVIEW

**Comment by zkochan** [comment]
I think using [require.resolve](https://nodejs.org/api/all.html#modules_require_resolve_request_options) to find the `node-gyp` location would also work

**Comment by zkochan** [comment]
oh, I just recalled, I moved `node-gyp` to npm-lifecycle. https://github.com/npm/lifecycle/issues/4

The node-gyp probably can be removed from supi altogether.

**Comment by zkochan** [comment]
Seems like `find-up` can be removed as well. And try to install `npm-lifecycle` instead of `@zkochan/npm-lifecycle`

They have all the changes pnpm needs. Just rename the module declaration at the end of https://github.com/pnpm/supi/blob/master/typings/local.d.ts and it should work

**Comment by etamponi** [comment]
Awesome :+1:

Il 24 dic 2017 21:18, "Zoltan Kochan" <notifications@github.com> ha scritto:

**Comment by etamponi** [comment]
All done! Merry Christmas :)

File: src/install/postInstall.ts [hunk 1]
```diff
@@ -114,21 +108,12 @@ export async function npmRunScript (
...
+  scripts: {},
 ) {
   try {
     await fs.stat(path.join(root, 'binding.gyp'))
+    scripts['install'] = 'node-gyp rebuild'
   } catch (err) {
-    if ((<NodeJS.ErrnoException>err).code === 'ENOENT') {
-      return
-    }
+    return
```
Comment (zkochan) [inline src/install/postInstall.ts]: return is not needed. You can even do `} catch {}`. If typescript will allow it

**Comment by zkochan** [comment]
Thanks! Merry Christmas to you as well! :christmas_tree:

**Review by zkochan — APPROVED** [review]

**Comment by etamponi** [comment]
squashed all commits into one

**Comment by zkochan** [comment]
ok, cool. I'll wait for the tests to pass and merge it

**Pull request merged**